src/shared/views/ChangeUserName.js

```diff
@@ -80,29 +89,45 @@ type InjectedProps = {
 type Props = OwnProps & InjectedProps;
 
 function ChangeUserName(props: Props) {
-    const [newUserName, setNewUserName] = React.useState();
+    const [newUserName, setNewUserName] = useState();
+    const [showUserNameError, setShowUserNameError] = useState(false);
+    const [updatePending, setUpdatePending] = useState(false);
 
     const { t, navigation, firebase } = props;
     const userName = firebase.auth().currentUser?.displayName;
-    const [updatePending, setUpdatePending] = React.useState(false);
-
-    const handleConfirmButtonClick = React.useCallback(() => {
-        if ((newUserName?.length ?? 0) >= MIN_USERNAME_LENGTH) {
-            setUpdatePending(true);
-            firebase
-                .updateAuth({ displayName: newUserName })
-                .then(() => firebase.updateProfile({ username: newUserName }))
-                .then(
-                    () => {
-                        setUpdatePending(false);
-                        navigation.goBack();
-                    },
-                    () => {
-                        setUpdatePending(false);
-                        console.error('failed to change update user profile');
-                    },
-                );
+
+    const handleConfirmButtonClick = React.useCallback(async () => {
+        const isValidUserName = validateUserName(newUserName);
+        if (!isValidUserName) {
+            setShowUserNameError(true);
```

Comment:
Why not just use MessageBarManager.showAlert?

Reply:
This is not consistent with the Login page.

Reply:
to  show error for already existed user in firebase we are using `MessageBarManager.showAlert`.
for user character validation we are showing error message as helper text

Reply:
This should at least be consistent with the other page.